server/stream.go

```diff
@@ -576,6 +582,16 @@ func (a *Account) addStreamWithAssignment(config *StreamConfig, fsConfig *FileSt
 	c := s.createInternalJetStreamClient()
 	ic := s.createInternalJetStreamClient()
 
+	// Work out the stream ingest limits.
+	mlen := s.opts.JetStreamMaxBufferMsgs
+	msz := uint64(s.opts.JetStreamMaxBufferSize)
```

Comment:
buffer size int but uint64 here will overflow?

Reply:
`int64` to `uint64`, overflow I suppose is possible but only over 9 million terabytes.

Reply:
ah got it confused with `JetStreamMaxBufferMsgs` above which is an int

Reply:
no wait, still if someone sets -1 on the int64 it will overflow here. Need to be explicit whether negative means unlimited or protect against that

Reply:
Yes, at the moment nothing stops you from setting an arbitrarily large limit normally, possibly even more than the system has available, so the negative underflow is really no worse than doing that.

Reply:
yeah good point

Reply:
so in that sense -1 underflow as a feature means that it is kind of infinite and disables the default